client/src/components/ratings-and-reviews/add-review-form/AddReviewForm.jsx

```diff
@@ -0,0 +1,143 @@
+import { React, useState } from 'react';
+import $ from 'jquery';
+import axios from 'axios';
+import CharacteristicSelection from './CharacteristicSelection';
+import PhotoUpload from './PhotoUpload';
+
+function AddReviewForm({
+  characteristics, showForm, modalHandler, productId,
+}) {
+  if (!showForm) {
+    return null;
+  }
+
+  const [reviewObj, setReviewObj] = useState({
+    product_id: productId,
+    rating: '',
+    summary: '',
+    body: '',
+    recommend: '',
+    name: '',
+    email: '',
+    photos: [],
+    characteristics: '',
+  });
+
+  function starHandler(e) {
+    const index = 5 - e.target.id;
+    $('.dynamic-star span').slice(0, index).css('color', '#ccc');
+    $('.dynamic-star span').slice(index).css('color', 'black');
+    setReviewObj({ ...reviewObj, rating: Number(e.target.id) });
+  }
+
+  function inputHandler(e, url) {
+    const characterArr = ['Size', 'Width', 'Comfort', 'Quality', 'Length', 'Fit'];
+    if (characterArr.includes(e.target.name)) {
+      const { id } = characteristics[e.target.name];
+      setReviewObj({
+        ...reviewObj,
+        characteristics: { ...reviewObj.characteristics, [id]: Number(e.target.value) },
+      });
+    } else if (e.target.name === 'photos') {
+      setReviewObj({ ...reviewObj, photos: reviewObj.photos.concat(url) });
```

Comment:
Does concat work here? The url parameter (reader.result) is presented here in the form of a string from readAndPreview()

Reply:
Nevermind! I mistakenly thought concat only takes array. We good.